File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int m = magazine.size(), n = ransomNote.size();
        if (m < n)
            return false;
        int cnt[26];
        for (char ch : magazine)
            cnt[ch-'a']++;
        for (char ch : ransomNote){
            if (--cnt[ch-'a'] < 0)
                return false;
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        n,m = len(ransomNote),len(magazine)
        if n > m: return False
        count = [0] * 26
        for ch in magazine:
            count[ord(ch)-ord('a')] += 1
        for ch in ransomNote:
            count[ord(ch)-ord('a')] -= 1
            if count[ord(ch)-ord('a')] < 0:
                return False
        return True  